hibernate-core/src/main/java/org/hibernate/mapping/Index.java

```diff
@@ -223,7 +224,22 @@ public String getName() {
 	}
 
 	public void setName(String name) {
-		this.name = name;
+		if (
+				StringHelper.isNotEmpty( name ) &&
+						Dialect.QUOTE.indexOf( name.charAt( 0 ) ) > -1 //TODO: deprecated, remove eventually
+				) {
+			quoted = true;
+			this.name = name.substring( 1, name.length() - 1 );
+		}
+		else {
+			this.name = name;
+		}
+	}
+
+	public String getQuotedName(Dialect d) {
+		return quoted ?
+				d.openQuote() + name + d.closeQuote() :
```

Comment:
Yes that is a general change we need to make all across `org.hibernate.mapping`.  We will focus on that in 7.  But we can certainly change little things like this piecemeal.